resources-generator/src/main/kotlin/dev/icerock/gradle/generator/StringsGenerator.kt

```diff
@@ -13,9 +13,36 @@ import dev.icerock.gradle.generator.jvm.JvmStringsGenerator
 import dev.icerock.gradle.utils.removeLineWraps
 import org.gradle.api.file.FileTree
 import java.io.File
+import java.util.Locale as JvmLocale
 import javax.xml.parsers.DocumentBuilderFactory
 
-typealias LanguageType = String
+sealed interface LanguageType {
+
+    object Base : LanguageType
+
+    class Locale(languageTag: String) : LanguageType {
+
+        val locale: JvmLocale = JvmLocale.forLanguageTag(languageTag)
+
+        override fun hashCode(): Int = locale.hashCode()
+        override fun toString(): String = locale.toLanguageTag()
```

Comment:
This is probably too implicit. We should expose `bcpString()` and `localeString()` functions.